boot/bootutil/src/bootutil_public.c

```diff
@@ -458,18 +458,18 @@ boot_swap_type_multi(int image_index)
     return BOOT_SWAP_TYPE_NONE;
 }
 
-static int flash_area_id_to_image(int id)
+static int flash_area_to_image(const struct flash_area *fa)
```

Comment:
Looks, OK
Only one my concern is that what if `fa`  would out of range mcuboot's images.

Reply:
Yeah, that is one of the things we should take care within code not to happen.

Reply:
Ok, I'm fine with that statement.